pkg/operator/controller/ttl_report.go

```diff
@@ -0,0 +1,87 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/starboard/pkg/operator/etc"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type TTLReportReconciler struct {
+	logr.Logger
+	etc.Config
+	client.Client
+}
+
+func (r *TTLReportReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.VulnerabilityReport{}).
+		Complete(r.reconcileReport())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *TTLReportReconciler) reconcileReport() reconcile.Func {
+	return func(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+		log := r.Logger.WithValues("report", req.NamespacedName)
+
+		report := &v1alpha1.VulnerabilityReport{}
+		err := r.Client.Get(ctx, req.NamespacedName, report)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.V(1).Info("Ignoring cached report that must have been deleted")
+				return ctrl.Result{}, nil
+			}
+			return ctrl.Result{}, fmt.Errorf("getting report from cache: %w", err)
+		}
+		ttlReportAnnotationStr := report.Annotations[v1alpha1.TTLReportAnnotation]
+		if ttlReportAnnotationStr != "" {
+			creationTime := report.GetCreationTimestamp()
```

Comment:
To persist vulnerability reports we do create or update so in general we should be looking at the UpdateTimestamp property. WDYT?

```suggestion
			creationTime := report.Report.UpdateTimestamp
```
BTW, in createOrUpdate method we do clone instance without rewriting annotations.

Reply:
Hadn't actually looked what UpdateTimestamp did. It sounds reasonable to change over to it.